functions/index.js

```diff
@@ -34,19 +39,40 @@ exports.inboundSMS = functions.https.onRequest(async (req, res) => {
 // This function listens for updates to the Firebase Realtime Database
 // and sends a message back to the original sender
 exports.sendSMS = functions.database
-  .ref('/msgq/{pushId}')
+  .ref("/msgq/{pushId}")
   .onCreate(async (message) => {
     const { msisdn, text, to } = message.val();
-    // the incoming object - 'msisdn' is the your phone number, and 'to' is the Vonage number
-    return vonage.sms.send(to, msisdn, `You sent the following text: ${text}`, (err, res) => {
-      if (err) {
-        console.log(err);
+
+    const resultSnapshot = await message.ref.parent
+      .child("result")
+      .once("value");
+    if (resultSnapshot.exists()) {
+      console.log("Result exists already.");
+      return;
+    }
+
+    try {
+      const responseData = await vonage.sms.send({
+        to: to,
+        from: msisdn,
+        text: `You sent the following text: ${text}`,
+      });
```

Comment:
I don't know this blog post, so my only comment is: If you have created an application, I suggest using the messages API instead of SMS. If this post does not require an application, this can stay as is

Reply:
So, when I got this suggestion some time ago, because this blog post is very converting, the suggestion was to create a new separate blog. So this is an old one I am updating to continue to use the SMS API but we do have another one using the Messages API: https://developer.vonage.com/en/blog/send-sms-messages-with-cloud-functions-for-firebase-gen-2